test/integration/services/alert.js

```diff
@@ -0,0 +1,76 @@
+describe('Alert Service', function() {
+  specify('Displaying', function() {
+    cy
+      .visit('/');
+
+    cy
+      .window()
+      .should('have.property', 'Radio')
+      .then(Radio => {
+        Radio.request('alert', 'show:success', 'success');
+      });
+
+    cy
+      .get('.alert-box')
+      .contains('success')
+      .wait(4800)
+      .should('not.exist');
```

Comment:
that work for you @paulfalgout?

Reply:
Try using cy.clock here to set the time and move it forward so that the test doesn't delay.

Reply:
also just to be sure this test is correct, try `should('exist')` to be sure it fails... sometimes the negative tests can be false passing because maybe `wait` returns something different than we expect or whatever.  _although_... just came to me.. the other place we could test this is with another test using the `onComplete` stub.  That'd be more reliable that `not.exist` come to think of it.

Reply:
I've tried various iterations of this:
```js
cy
  .clock()
  .get('.alert-box')
  .contains('success')
  .should('exist')
  .then(() => {
    this.clock.tick(4800);
  })
  // or just tick()
  .should('not.exist');
```

but no dice. Thoughts on what I'm missing?

Reply:
the clock is mocking the javascript timer api, so the clock / tick needs to surround the method building the timeout.. so it'd be more like:
```
clock()  // prevent any timers from moving on their own
window()
...
Radio => {
 request('modal')  // creates timer
}
.should('exist')  // timer already created but no "time" has passed
tick(n)  // pass the time
.should('not.exist')

clock.restore()
```